### PR TITLE
Add support for `PackedVector4Array`

### PR DESCRIPTION
**Pull request by dsnopek**
Fixes https://github.com/godotengine/godot-cpp/issues/1454

Adds support for `PackedVector4Array` from Godot PR https://github.com/godotengine/godot/pull/85474

### CODE REVIEW

File: src/variant/packed_arrays.cpp [hunk 1]
```diff
@@ -198,6 +199,24 @@ Vector3 *PackedVector3Array::ptrw() {
 	return (Vector3 *)internal::gdextension_interface_packed_vector3_array_operator_index((GDExtensionTypePtr *)this, 0);
 }
 
+const Vector4 &PackedVector4Array::operator[](int64_t p_index) const {
+	const Vector4 *vec = (const Vector4 *)internal::gdextension_interface_packed_vector4_array_operator_index_const((GDExtensionTypePtr *)this, p_index);
+	return *vec;
+}
+
+Vector4 &PackedVector4Array::operator[](int64_t p_index) {
+	Vector4 *color = (Vector4 *)internal::gdextension_interface_packed_vector4_array_operator_index((GDExtensionTypePtr *)this, p_index);
```
Comment (AThousandShips) [inline src/variant/packed_arrays.cpp:208]:
```suggestion
	Vector4 *vec = (Vector4 *)internal::gdextension_interface_packed_vector4_array_operator_index((GDExtensionTypePtr *)this, p_index);
```

Comment (dsnopek) [reply src/variant/packed_arrays.cpp]: Whoops! Copy-paste error :-) Thanks for catching it!

**Review by AThousandShips — APPROVED** [review]
LGTM

**Comment by YuriSizov** [comment]
> Godot `master` is a moving target, and godot-cpp `master` needs to change to keep up.

Yes, I understand that. But even so, in the engine repo one way changes like that are marked as breaking compat. Although I see that the label is not really used here. It would still be helpful to know when such changes occur.

> Anyway, I guess if we wanted to start worrying about compatibility for -dev releases, we could keep track of commit hashes like we do for -beta and -rc releases?

My main point was having information. I have no idea if it's feasible or reasonable to worry about backwards compatibility with such changes. Right now you find out that the commit is incompatible by trying it and building it and trying to run it, which is more time consuming than checking for a label on recently merged PRs 🙃

**Pull request merged**